shunting-yard-common/src/test/java/com/hotels/shunting/yard/common/io/jackson/DeserializerWithJsonInputTest.java

```diff
@@ -61,71 +73,104 @@
-  private final static String DROP_PARTITION_EVENT = "\"Message\" : \"{\\\"protocolVersion\\\":\\\"1.0\\\",\\\"eventType\\\":\\\"DROP_PARTITION\\\",\\\"dbName\\\":\\\"some_db\\\",\\\"tableName\\\":\\\"some_table\\\",\\\"partitionKeys\\\": {\\\"col_1\\\": \\\"string\\\", \\\"col_2\\\": \\\"integer\\\",\\\"col_3\\\": \\\"string\\\"},\\\"partitionValues\\\":[\\\"val_1\\\", \\\"val_2\\\", \\\"val_3\\\"]}\"";
-
-  private final static String CREATE_TABLE_EVENT = "\"Message\" : \"{\\\"protocolVersion\\\":\\\"1.0\\\",\\\"eventType\\\":\\\"CREATE_TABLE\\\",\\\"dbName\\\":\\\"some_db\\\",\\\"tableName\\\":\\\"some_table\\\"}\"";
-  private final static String DROP_TABLE_EVENT = "\"Message\" : \"{\\\"protocolVersion\\\":\\\"1.0\\\",\\\"eventType\\\":\\\"DROP_TABLE\\\",\\\"dbName\\\":\\\"some_db\\\",\\\"tableName\\\":\\\"some_table\\\"}\"";
-  private final static String INSERT_EVENT = "\"Message\" : \"{\\\"protocolVersion\\\": \\\"1.0\\\",\\\"eventType\\\": \\\"INSERT\\\",\\\"dbName\\\": \\\"some_db\\\",\\\"tableName\\\": \\\"some_table\\\",\\\"files\\\": [\\\"file:/a/b.txt\\\",\\\"file:/a/c.txt\\\"],\\\"fileChecksums\\\": [\\\"123\\\",\\\"456\\\"],\\\"partitionKeyValues\\\": {\\\"col_1\\\": \\\"val_1\\\",\\\"col_2\\\": \\\"val_2\\\", \\\"col_3\\\": \\\"val_3\\\"}}\"";
+  private String addPartitionEvent;
+  private String alterPartitionEvent;
+  private String createTableEvent;
+  private String dropPartitionEvent;
+  private String insertEvent;
+  private String alterTableEvent;
+  private String dropTableEvent;
 
   private static final Map<String, String> PARTITION_KEYS_MAP = ImmutableMap
       .of("col_1", "string", "col_2", "integer", "col_3", "string");
   private static final List<String> PARTITION_VALUES = ImmutableList.of("val_1", "val_2", "val_3");
   private static final List<String> OLD_PARTITION_VALUES = ImmutableList.of("val_4", "val_5", "val_6");
+
+  private static final Map<String, String> TABLE_PARAM_MAP = ImmutableMap.of("param_1", "val_1", "param_2", "val_2");
+
   private final static String TEST_DB = "some_db";
   private final static String TEST_TABLE = "some_table";
+  private final static String TEST_TABLE_LOCATION = "s3://table_location";
+  private final static String OLD_TEST_TABLE_LOCATION = "s3://old_table_location";
+  private final static String PARTITION_LOCATION = "s3://table_location/partition_location";
+  private final static String OLD_PARTITION_LOCATION = "s3://table_location/old_partition_location";
+
+  @Before
+  public void init() throws IOException {
+    addPartitionEvent = new String(Files.readAllBytes(dataFolder.getFile("add_partition.json").toPath()))
+        .replace("\n", "");
```

Comment:
why do you need to replace '\n'?

Reply:
coz its not a valid json string with '\n' characters in it.

Reply:
What error do you get (and from where) if you don't remove the '\n'? Just curious, feels like we should be able to handle human readable JSON, maybe there's just a setting we could use when we configure the mapper or something. If not then we'll have to live with this.

Reply:
Found this to fix this issue:

`mapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_CONTROL_CHARS, true)`